tools/cldr-code/src/main/java/org/unicode/cldr/util/XMLFileReader.java

```diff
@@ -416,6 +425,7 @@ public void startElement(
             startElements.push(tempPath.toString());
             chars.setLength(0); // clear garbage
             lastIsStart = true;
+            simpleHandler.handleElement(tempPath);
```

Comment:
It's a new API.  There wasn't a way (in this API) to register for notification on a new element.  The issue comes because the data is of this form:

```xml
<a>
   <b c="3"/>
   <b d="4"/>
</a>
<a>
   <b c="5"/>
   <b d="6"/>
</a>
```

the XPATH=based api would only produce:

```
//a/b[@c="3"]
//a/b[@d="4"]
 (need notification that we're on a new 'a' here)
//a/b[@c="5"]
//a/b[@c="6"]
```

since there's no identity on `<a>` this new API lets me reset the parser.  Anyway, that's what it's for.

Reply:
I assume this is a special exception for keyboards, right? For normal CLDR there must be a distinguishing attribute for the first a and the second a to be different.

In CLDR, the mechanism for such distinctions is to add an _q distinguishing element. That also preserves order where needed. Why wasn't that done here?

Reply:
@macchiati this is the internal low-level XMLFileReader interface. _q is handled at the next layer up.

Reply:
normal CLDR data doesn't go through this interface, see other comment.